patches/third_party-blink-renderer-core-dom-document.cc.patch

```diff
@@ -1,13 +1,13 @@
 diff --git a/third_party/blink/renderer/core/dom/document.cc b/third_party/blink/renderer/core/dom/document.cc
-index 01f4b1d7ffe928126be221ff39715d1c0e6ae849..14d5d0201a17e78765788584f06e589b4215b743 100644
+index 01f4b1d7ffe928126be221ff39715d1c0e6ae849..5c0f9a34f1337e2848345db63ef306b71b38a044 100644
 --- a/third_party/blink/renderer/core/dom/document.cc
 +++ b/third_party/blink/renderer/core/dom/document.cc
 @@ -33,7 +33,7 @@
  // instead of including more headers. If that is infeasible, adjust the limit.
  // For more info, see
  // https://chromium.googlesource.com/chromium/src/+/HEAD/docs/wmax_tokens.md
 -#pragma clang max_tokens_here 900000
-+#pragma clang max_tokens_here 2000000
++#pragma clang max_tokens_here 2097152
```

Comment:
we really need to look at this because I can't imagine why we need this to be an order of magnitude higher than upstream. There is something wrong here and it needs to be fixed

Reply:
Yes, I don't have a good explanation for this just yet, but I can tell that so far this has been **not** a problem on the cr88 branch, so maybe it was something temporary? I'll keep an eye on it.

Reply:
maybe this is something that needs to be fixed in our code? https://chromium.googlesource.com/chromium/src/+/HEAD/docs/wmax_tokens.md#what-to-do-when-hitting-a-wmax_tokens-warning

Reply:
From what I understand, it feels like Brave's code could be abusing a bit too much of using `#include` everywhere instead of the current policy upstream of adding forward declarations in header files + only using the `#include` directive from implementation files, whenever possible.

It's hard to know without actually auditing the code and actually make changes, though. What would you think of filing a github issue to "Use forward declarations in header files whenever possible" to track the needed work to move Brave code into that direction? Even if we don't ever manage to get it building with a limit of 900K tokens, it would still be worth IMHO to lower it as much as possible from the current massive limit of 2M!

Reply:
I wonder if this is because the rust code on linux uses the wrong sysroot

Reply:
Hmmm... actually that could be it (i.e. an environment-based issue), because I just tried removing this change on my cr88 branch and leaving the upstream default of `900000` and it just built fine. And lowering the pragma a bit (e.g. to `800000`) I can see it failing because `the number of preprocessor source tokens (875577) exceeds this token limit (800000)`.

So, it seems this was maybe a punctual problem at some point on cr86 and cr87 (cr86 was the first time I saw it, and then in cr87 again), but since it does not happen anymore on cr88 maybe it's not worth investigating further and we should simply drop it from cr88. WDYT?